iptables/iptables_test.go

```diff
@@ -66,45 +66,46 @@ func contains(list []string, value string) bool {
 	return false
 }
 
-// Create an array of IPTables with different hasWait/hasCheck to
-// test different behaviours
-func mustTestableIptables() []*IPTables {
+// mustTestableIptables returns a list of ip(6)tables handles with various
+// features enabled & disabled, to test compatability.
+// We used to test noWait as well, but that was removed as of iptables v1.6.0
+func mustTestableIptables(t *testing.T) []*IPTables {
 	ipt, err := New()
 	if err != nil {
-		panic(fmt.Sprintf("New failed: %v", err))
+		t.Fatal(fmt.Sprintf("New failed: %v", err))
 	}
 	ip6t, err := NewWithProtocol(ProtocolIPv6)
 	if err != nil {
-		panic(fmt.Sprintf("NewWithProtocol(ProtocolIPv6) failed: %v", err))
+		t.Fatal(fmt.Sprintf("NewWithProtocol(ProtocolIPv6) failed: %v", err))
 	}
 	ipts := []*IPTables{ipt, ip6t}
-	// ensure we check one variant without built-in locking
-	if ipt.hasWait {
-		iptNoWait := &IPTables{
-			path:    ipt.path,
-			hasWait: false,
-		}
-		ipts = append(ipts, iptNoWait)
-	}
+
 	// ensure we check one variant without built-in checking
 	if ipt.hasCheck {
-		iptNoCheck := &IPTables{
-			path:     ipt.path,
-			hasCheck: false,
-		}
-		ipts = append(ipts, iptNoCheck)
+		i := *ipt
+		i.hasCheck = false
+		ipts = append(ipts, &i)
+
+		i6 := *ip6t
+		i6.hasCheck = false
+		ipts = append(ipts, &i6)
+	} else {
+		t.Fatal("iptables on this machine is too old -- missing -C")
 	}
 	return ipts
 }
 
 func TestChain(t *testing.T) {
-	for _, ipt := range mustTestableIptables() {
+	for _, ipt := range mustTestableIptables(t) {
 		runChainTests(t, ipt)
 	}
 }
 
 func runChainTests(t *testing.T, ipt *IPTables) {
-	t.Logf("testing %s (hasWait=%t, hasCheck=%t)", getIptablesCommand(ipt.Proto()), ipt.hasWait, ipt.hasCheck)
+	t.Logf("testing %s (hasWait=%t, hasCheck=%t)", ipt.path, ipt.hasWait, ipt.hasCheck)
+	if !ipt.hasWait {
```

Comment:
I guess you can completely get rid of `hasWait` in here and move the panic-check to `mustTestableIptables` if you now require `--wait`.